RK3399 Typec DP

Tonymac32 opened the pull request:
# Description

Type-C DP support for the NanoPC T4 and beginning DP support for the RockPi 4C.

The 4C has a DP plug with no Type-C controller, current drivers require an extcon of sorts, Jagan has addressed this by making a virtual power delivery driver that fills that role.  As I don't have the hardware I cannot test, but I also don't want to let this stagnate. Needs testing.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ X ] NanoPC T4 tested dual monitor configuration with HDMI + Type-C DP.
-  RockPi 4C needs tested and feedback provided concerning the capability/lack of the driver

# Checklist:

- [ X ] My code follows the style guidelines of this project
- [ X ] I have performed a self-review of my own code
- [ X ] I have commented my code, particularly in hard-to-understand areas
- [ X ] I have made corresponding changes to the documentation
- [ X ] My changes generate no new warnings
- [ X ] Any dependent changes have been merged and published in downstream modules

igorpecovnik commented:
http://ix.io/2Rsr

Attached HDMI only, screen OK
Attached HDMI + DP (4K), screen not OK, DP nothing
Attaching DP
[  376.327740] cdn-dp fec00000.dp: [drm:cdn_dp_pd_event_work [rockchipdrm]] Connected, not enabled. Enabling cdn
screen remain dark

Tonymac32 commented:
@igorpecovnik I've reverted the RockPi 4C one for now, I added a .WIP to the filename so it can be looked at by others.  Would like to merge this one for the NanoPC T4
